apache/metron
# METRON-2019: Improve Metron REST Logging

## Contributor Comments
This PR adds the logging improvements listed in the Jira.

### Changes Included

- Updated the metron-rest pom.xml file to use Log4j2 with SLF4J.
- Added an exception handler to properly log 400 errors
- Removed old log4j.properties files
- Added classes for logging Http request and response information
- Documentation for configuring log levels added to the REST README

### Testing
This has been tested in full dev.

1. Spin up full dev and navigate to Swagger
2. Monitor the REST log by logging into full dev and tailing the log:
```
tail -f /var/log/metron/metron-rest.log
```
3. Logging should appear to be set at ERROR level initially.
4. Verify 400 errors are now properly logged by sending a bad request:
```
curl -X POST --header 'Content-Type: application/json' --header 'Accept: */*' -d '{
  "filterClassName": {}
}' 'http://user:password@node1:8082/api/v1/sensor/parser/config/test'
```
5. Navigate to Ambari > Services > Metron > Configs > REST and locate the "Metron Spring options" setting.  Change it to `--logging.level.org.apache.metron.rest=INFO --logging.level.org.springframework.web.filter.CommonsRequestLoggingFilter=DEBUG --logging.level.org.apache.metron.rest.web.filter.ResponseLoggingFilter=DEBUG` and restart REST.
6. As the application restarts, there should be INFO level logging for Metron REST code.
7. Verify Http request and response information is logged by sending a request:
```
curl -X GET --header 'Accept: application/json' 'http://user:password@node1:8082/api/v1/global/config'
```


## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
